# Process Analysis Homework
For this assignment you will be reviewing a memory image infected with APT malware. The image has been provided for you to download and is named `APT.img` For this assignment we will just be focused on reviwing processes and determining if any of them need more detailed analysis. First you must identify what profile to use with your analysis with the `imageinfo` command

## Profile Identification

In [1]:
pip install yara

     -------------------------------------- 387.2/387.2 kB 3.4 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Created wheel for yara: filename=yara-1.7.7-py3-none-any.whl size=90782 sha256=4c7250289fa56083a746eab212e655bdaad9bfed5014db0c637bc4a06b04ba0f
  Stored in directory: c:\users\qrbbb\appdata\local\pip\cache\wheels\cc\f8\25\97bad16a22c87448f2145e87e1c41042f9172345d09072fe28
Successfully built yara


In [2]:
python ../../volatility3/vol.py -h

Volatility 3 Framework 2.5.2
usage: volatility [-h] [-c CONFIG] [--parallelism [{processes,threads,off}]]
                  [-e EXTEND] [-p PLUGIN_DIRS] [-s SYMBOL_DIRS] [-v] [-l LOG]
                  [-o OUTPUT_DIR] [-q] [-r RENDERER] [-f FILE]
                  [--write-config] [--save-config SAVE_CONFIG] [--clear-cache]
                  [--cache-path CACHE_PATH] [--offline]
                  [--single-location SINGLE_LOCATION]
                  [--stackers [STACKERS ...]]
                  [--single-swap-locations [SINGLE_SWAP_LOCATIONS ...]]
                  plugin ...

An open-source memory forensics framework

optional arguments:
  -h, --help            Show this help message and exit, for specific plugin
                        options use 'volatility <pluginname> --help'
  -c CONFIG, --config CONFIG
                        Load the configuration from a json file
  --parallelism [{processes,threads,off}]
                        Enables parallelism (defaults to off if no argum

In [2]:
python ../../volatility3/vol.py -vv

INFO     volatility3.cli: Volatility plugins path: ['C:\\Users\\qrbbb\\OneDrive\\Documents\\cybs\\volatility3\\volatility3\\plugins', 'C:\\Users\\qrbbb\\OneDrive\\Documents\\cybs\\volatility3\\volatility3\\framework\\plugins']
INFO     volatility3.cli: Volatility symbols path: ['C:\\Users\\qrbbb\\OneDrive\\Documents\\cybs\\volatility3\\volatility3\\symbols', 'C:\\Users\\qrbbb\\OneDrive\\Documents\\cybs\\volatility3\\volatility3\\framework\\symbols']
usage: volatility [-h] [-c CONFIG] [--parallelism [{processes,threads,off}]]
                  [-e EXTEND] [-p PLUGIN_DIRS] [-s SYMBOL_DIRS] [-v] [-l LOG]
                  [-o OUTPUT_DIR] [-q] [-r RENDERER] [-f FILE]
                  [--write-config] [--save-config SAVE_CONFIG] [--clear-cache]
                  [--cache-path CACHE_PATH] [--offline]
                  [--single-location SINGLE_LOCATION]
                  [--stackers [STACKERS ...]]
                  [--single-swap-locations [SINGLE_SWAP_LOCATIONS ...]]
                  plu

Error: Command failed: SubmitCode: python ../../volatility3/vol.py -vv

In [3]:
conda install pefile

DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): repo.anaconda.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): repo.anaconda.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): repo.anaconda.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): repo.anaconda.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): repo.anaconda.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): repo.anaconda.com:443
DEBUG:urllib3.connectionpool:https://repo.anaconda.com:443 "GET /pkgs/main/win-64/current_repodata.json HTTP/1.1" 304 0
DEBUG:urllib3.connectionpool:https://repo.anaconda.com:443 "GET /pkgs/r/win-64/current_repodata.json HTTP/1.1" 304 0
DEBUG:urllib3.connectionpool:https://repo.anaconda.com:443 "GET /pkgs/main/noarch/current_repodata.json HTTP/1.1" 304 0
DEBUG:urllib3.connectionpool:https://repo.anaconda.com:443 "GET /pkgs/msys2/noarch/current_repodata.json HTTP/1.1" 304 0
DEBUG:

In [4]:
python ../../volatility3/vol.py -q -f C:\Users\qrbbb\Downloads\APT.img windows.info.Info

Volatility 3 Framework 2.5.2

Variable	Value

Kernel Base	0x804d7000
DTB	0x319000
Symbols	file:///C:/Users/qrbbb/OneDrive/Documents/cybs/volatility3/volatility3/symbols/windows/ntkrnlpa.pdb/FD7B256CD2D14BC0B35BA9B89F691346-1.json.xz
Is64Bit	False
IsPAE	True
layer_name	0 WindowsIntelPAE
memory_layer	1 FileLayer
KdDebuggerDataBlock	0x80545b60
NTBuildLab	2600.xpsp_sp3_gdr.080814-1236
CSDVersion	3
KdVersionBlock	0x80545b38
Major/Minor	15.2600
MachineType	332
KeNumberProcessors	1
SystemTime	2009-05-05 19:28:57
NtSystemRoot	C:\WINDOWS
NtProductType	NtProductWinNt
NtMajorVersion	5
NtMinorVersion	1
PE MajorOperatingSystemVersion	5
PE MinorOperatingSystemVersion	1
PE Machine	332
PE TimeDateStamp	Thu Aug 14 09:33:12 2008


Now that you know the profile use that to review the following process analysis with volatility:
 * `plist`
 * `pstree`
 * `psscan`
 
 Document any of your findings of supicious processes in line and make sure to document each analysis execution.
 
 ## PSLIST

In [5]:
python ../../volatility3/vol.py -q -f C:\Users\qrbbb\Downloads\APT.img windows.pslist.PsList

Volatility 3 Framework 2.5.2

PID	PPID	ImageFileName	Offset(V)	Threads	Handles	SessionId	Wow64	CreateTime	ExitTime	File output

4	0	System	0x823c8830	55	254	N/A	False	N/A	N/A	Disabled
564	4	smss.exe	0x8230aad8	3	19	N/A	False	2009-04-16 16:10:01.000000 	N/A	Disabled
636	564	csrss.exe	0x822ca2c0	10	356	0	False	2009-04-16 16:10:06.000000 	N/A	Disabled
660	564	winlogon.exe	0x81f63020	16	502	0	False	2009-04-16 16:10:06.000000 	N/A	Disabled
704	660	services.exe	0x81f22020	15	254	0	False	2009-04-16 16:10:06.000000 	N/A	Disabled
716	660	lsass.exe	0x82164da0	21	342	0	False	2009-04-16 16:10:06.000000 	N/A	Disabled
872	704	vmacthlp.exe	0x822cb458	1	25	0	False	2009-04-16 16:10:07.000000 	N/A	Disabled
884	704	svchost.exe	0x81e54da0	17	208	0	False	2009-04-16 16:10:07.000000 	N/A	Disabled
968	704	svchost.exe	0x81da4590	10	241	0	False	2009-04-16 16:10:07.000000 	N/A	Disabled
1088	704	svchost.exe	0x81f739b0	70	1445	0	False	2009-04-16 16:10:07.000000 	N/A	Disabled
1140	704	svchost.exe	0x8232c020	5	60	0	

In [8]:
python ../../volatility3/vol.py -q -r csv -f C:\Users\qrbbb\Downloads\APT.img windows.pslist.PsList | csvcut -c 2,3,4,6,7,10,11 | csvlook

Volatility 3 Framework 2.5.2
|   PID |  PPID | ImageFileName   | Threads | Handles |          CreateTime | ExitTime |
| ----- | ----- | --------------- | ------- | ------- | ------------------- | -------- |
|     4 |     0 | System          |      55 |     254 |                     |          |
|   564 |     4 | smss.exe        |       3 |      19 | 2009-04-16 16:10:01 |          |
|   636 |   564 | csrss.exe       |      10 |     356 | 2009-04-16 16:10:06 |          |
|   660 |   564 | winlogon.exe    |      16 |     502 | 2009-04-16 16:10:06 |          |
|   704 |   660 | services.exe    |      15 |     254 | 2009-04-16 16:10:06 |          |
|   716 |   660 | lsass.exe       |      21 |     342 | 2009-04-16 16:10:06 |          |
|   872 |   704 | vmacthlp.exe    |       1 |      25 | 2009-04-16 16:10:07 |          |
|   884 |   704 | svchost.exe     |      17 |     208 | 2009-04-16 16:10:07 |          |
|   968 |   704 | svchost.exe     |      10 |     241 | 2009-04-16 16:10:07 |    

## PSTREE

In [9]:
python ../../volatility3/vol.py -q -f C:\Users\qrbbb\Downloads\APT.img windows.pstree.PsTree

Volatility 3 Framework 2.5.2

PID	PPID	ImageFileName	Offset(V)	Threads	Handles	SessionId	Wow64	CreateTime	ExitTime

4	0	System	0x823c8830	55	254	N/A	False	N/A	N/A
* 564	4	smss.exe	0x8230aad8	3	19	N/A	False	2009-04-16 16:10:01.000000 	N/A
** 660	564	winlogon.exe	0x81f63020	16	502	0	False	2009-04-16 16:10:06.000000 	N/A
*** 704	660	services.exe	0x81f22020	15	254	0	False	2009-04-16 16:10:06.000000 	N/A
**** 1088	704	svchost.exe	0x81f739b0	70	1445	0	False	2009-04-16 16:10:07.000000 	N/A
***** 1260	1088	wscntfy.exe	0x81f96220	1	39	0	False	2009-04-16 16:10:22.000000 	N/A
**** 872	704	vmacthlp.exe	0x822cb458	1	25	0	False	2009-04-16 16:10:07.000000 	N/A
**** 968	704	svchost.exe	0x81da4590	10	241	0	False	2009-04-16 16:10:07.000000 	N/A
**** 1512	704	spoolsv.exe	0x8219b630	10	129	0	False	2009-04-16 16:10:10.000000 	N/A
**** 1032	704	VMwareService.e	0x81dc2570	3	175	0	False	2009-04-16 16:10:16.000000 	N/A
**** 464	704	alg.exe	0x81d33628	6	105	0	False	2009-04-16 16:10:21.000000 	N/A
**** 1140	704	

## PSSCAN

In [10]:
python ../../volatility3/vol.py -q -f C:\Users\qrbbb\Downloads\APT.img windows.psscan.PsScan

Volatility 3 Framework 2.5.2

PID	PPID	ImageFileName	Offset(V)	Threads	Handles	SessionId	Wow64	CreateTime	ExitTime	File output

464	704	alg.exe	0x1f33628	6	105	0	False	2009-04-16 16:10:21.000000 	N/A	Disabled
968	704	svchost.exe	0x1fa4590	10	241	0	False	2009-04-16 16:10:07.000000 	N/A	Disabled
1672	1624	explorer.exe	0x1fa71a8	15	586	0	False	2009-04-16 16:10:10.000000 	N/A	Disabled
796	884	iexplore.exe	0x1fbdda0	8	152	0	False	2009-05-05 19:28:28.000000 	N/A	Disabled
2004	1672	VMwareUser.exe	0x1fc1a78	8	228	0	False	2009-04-16 16:10:11.000000 	N/A	Disabled
1032	704	VMwareService.e	0x1fc2570	3	175	0	False	2009-04-16 16:10:16.000000 	N/A	Disabled
840	1672	cmd.exe	0x204d648	1	33	0	False	2009-05-05 15:56:24.000000 	N/A	Disabled
884	704	svchost.exe	0x2054da0	17	208	0	False	2009-04-16 16:10:07.000000 	N/A	Disabled
1212	704	svchost.exe	0x2091da0	14	208	0	False	2009-04-16 16:10:09.000000 	N/A	Disabled
2020	1672	ctfmon.exe	0x211a650	1	71	0	False	2009-04-16 16:10:11.000000 	N/A	Disabled
1984	1672	V

In [11]:
python ../../volatility3/vol.py -q -r csv -f C:\Users\qrbbb\Downloads\APT.img windows.psscan.PsScan | csvcut -c 2,3,4,6,7,10,11 | csvlook

Volatility 3 Framework 2.5.2
|   PID |  PPID | ImageFileName   | Threads | Handles |          CreateTime | ExitTime |
| ----- | ----- | --------------- | ------- | ------- | ------------------- | -------- |
|   464 |   704 | alg.exe         |       6 |     105 | 2009-04-16 16:10:21 |          |
|   968 |   704 | svchost.exe     |      10 |     241 | 2009-04-16 16:10:07 |          |
| 1,672 | 1,624 | explorer.exe    |      15 |     586 | 2009-04-16 16:10:10 |          |
|   796 |   884 | iexplore.exe    |       8 |     152 | 2009-05-05 19:28:28 |          |
| 2,004 | 1,672 | VMwareUser.exe  |       8 |     228 | 2009-04-16 16:10:11 |          |
| 1,032 |   704 | VMwareService.e |       3 |     175 | 2009-04-16 16:10:16 |          |
|   840 | 1,672 | cmd.exe         |       1 |      33 | 2009-05-05 15:56:24 |          |
|   884 |   704 | svchost.exe     |      17 |     208 | 2009-04-16 16:10:07 |          |
| 1,212 |   704 | svchost.exe     |      14 |     208 | 2009-04-16 16:10:09 |    

## Summary of Findings
Please summarize all your findings here with cleaned up documented evidence

***** 796	884	iexplore.exe	0x81dbdda0	8	152	0	False	2009-05-05 19:28:28.000000 	N/A
this process is supicious it start much later than its parent process